packages/medusa/src/repositories/product.ts

```diff
@@ -376,6 +372,10 @@ export class ProductRepository extends Repository<Product> {
       delete where.title
     }
 
+    if ("price_list_id" in where) {
+      delete where?.price_list_id
```

Comment:
This would break the convention followed in the rest of the method body, which is why I went with this approach

Reply:
You can read the details of the bug in the issues in the description :)

Reply:
this approach can be problematic, and this is why we don't do that anymore normally. by modifying the object, you also modify the object that the consumer has passed, so, if the consumer want to use that object for something else, it can lead to some problem since the properties does not exists anymore. @srindom wdyt?

Reply:
we should always returned the mutated object and not mutate the input object, except if this is the purpose, 
here, the consumer of `getFreeTextSearchResultsAndCount` does not expect his options to be mutated no?

Reply:
it is not problematic at first glance but can quickly become, specially if we reuse the object elsewhere without noticing that it has been mutated, like we faced in other pr's. also, in other cases, this is the way to hold the references everywhere and create an expected memory consumption because the objects cannot be entirely collected since some references are here and there.